sim/midas/src/main/scala/midas/platform/F1Shim.scala

```diff
@@ -5,240 +5,38 @@ import chisel3._
-      printf("[dma,arfire,%x] addr %x, len %x, size %x, burst %x, lock %x, cache %x, prot %x, qos %x, region %x, id %x, user %x\n",
-        cyclecount,
-        io.dma.ar.bits.addr,
-        io.dma.ar.bits.len,
-        io.dma.ar.bits.size,
-        io.dma.ar.bits.burst,
-        io.dma.ar.bits.lock,
-        io.dma.ar.bits.cache,
-        io.dma.ar.bits.prot,
-        io.dma.ar.bits.qos,
-        io.dma.ar.bits.region,
-        io.dma.ar.bits.id,
-        io.dma.ar.bits.user)
-    }
-
-    when (io.dma.r.fire()) {
-      printf("[dma,rfire,%x] resp %x, data %x, last %x, id %x, user %x\n",
-        cyclecount,
-        io.dma.r.bits.resp,
-        io.dma.r.bits.data,
-        io.dma.r.bits.last,
-        io.dma.r.bits.id,
-        io.dma.r.bits.user)
+    if (p(AXIDebugPrint)) {
+      AXI4Printf(io.master, "master")
+      AXI4Printf(io.dma,    "dma")
+      io_slave.zipWithIndex foreach { case (io, idx) => AXI4Printf(io, s"slave_${idx}") }
     }
 
-  when (io.slave(0).aw.fire()) {
-    printf("[slave,awfire,%x] addr %x, len %x, size %x, burst %x, lock %x, cache %x, prot %x, qos %x, region %x, id %x, user %x\n",
-      cyclecount,
-
-      io.slave(0).aw.bits.addr,
-      io.slave(0).aw.bits.len,
-      io.slave(0).aw.bits.size,
-      io.slave(0).aw.bits.burst,
-      io.slave(0).aw.bits.lock,
-      io.slave(0).aw.bits.cache,
-      io.slave(0).aw.bits.prot,
-      io.slave(0).aw.bits.qos,
-      io.slave(0).aw.bits.region,
-      io.slave(0).aw.bits.id,
-      io.slave(0).aw.bits.user
-      )
-  }
-
-  when (io.slave(0).w.fire()) {
-    printf("[slave(0),wfire,%x] data %x, last %x, id %x, strb %x, user %x\n",
-      cyclecount,
-
-      io.slave(0).w.bits.data,
-      io.slave(0).w.bits.last,
-      io.slave(0).w.bits.id,
-      io.slave(0).w.bits.strb,
-      io.slave(0).w.bits.user
-      )
-  }
-
-  when (io.slave(0).b.fire()) {
-    printf("[slave(0),bfire,%x] resp %x, id %x, user %x\n",
-      cyclecount,
-
-      io.slave(0).b.bits.resp,
-      io.slave(0).b.bits.id,
-      io.slave(0).b.bits.user
-      )
-  }
-
-  when (io.slave(0).ar.fire()) {
-    printf("[slave(0),arfire,%x] addr %x, len %x, size %x, burst %x, lock %x, cache %x, prot %x, qos %x, region %x, id %x, user %x\n",
-      cyclecount,
-
-      io.slave(0).ar.bits.addr,
-      io.slave(0).ar.bits.len,
-      io.slave(0).ar.bits.size,
-      io.slave(0).ar.bits.burst,
-      io.slave(0).ar.bits.lock,
-      io.slave(0).ar.bits.cache,
-      io.slave(0).ar.bits.prot,
-      io.slave(0).ar.bits.qos,
-      io.slave(0).ar.bits.region,
-      io.slave(0).ar.bits.id,
-      io.slave(0).ar.bits.user
-      )
-  }
+    top.module.io.ctrl <> io.master
+    top.module.io.dma  <> io.dma
+    io_slave.zip(top.module.mem).foreach({ case (io, bundle) => io <> bundle })
 
-  when (io.slave(0).r.fire()) {
-    printf("[slave(0),rfire,%x] resp %x, data %x, last %x, id %x, user %x\n",
-      cyclecount,
+    val (wCounterValue, wCounterWrap) = Counter(io.master.aw.fire(), 4097)
```

Comment:
I suppose this is old, but why exactly is it 4097 and not 4096?

Reply:
@sagark

Reply:
uhhhhh. probably just padding to avoid having to think about wraparound

Reply:
wait nvm